src/Components/Components/src/Rendering/ParameterViewLifetime.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Components.Rendering
+{
+    internal readonly struct ParameterViewLifetime
+    {
+        private readonly RenderBatchBuilder _owner;
+        private readonly int _stamp;
+
+        public static readonly ParameterViewLifetime Unbound = default;
+
+        public ParameterViewLifetime(RenderBatchBuilder owner)
+        {
+            _owner = owner;
+            _stamp = owner.ParameterViewValidityStamp;
+        }
+
+        public void AssertNotExpired()
+        {
+            // _owner will be null if this instance is default(ParameterViewLifetime)
+            if (_owner != null && _owner.ParameterViewValidityStamp != _stamp)
```

Comment:
Is it valid to call `ParameterView.GetEnumerator` on an unbound item? Should this complain if `_owner is null`?

Reply:
No, this behavior is desired, though I'll clarify the comment. The `_owner` being null implies the lifetime is unbounded, meaning that it's never going to expire.